### PR TITLE
OHRM5X-1261 : Prepare migrations classes for 4.3.3 to 4.10

### PR DESCRIPTION
**Pull request by SharukaNP**

### CODE REVIEW

File: installer/Migration/V4_10_0/Migration.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+namespace OrangeHRM\Installer\Migration\V4_10_0;
+
+class Migration extends \OrangeHRM\Installer\Util\V1\AbstractMigration
+{
+    /**
+     * @inheritDoc
+     */
+    public function up(): void
+    {
+        // TODO: Implement up() method.
```
Comment (RajithaKumara) [inline installer/Migration/V4_10_0/Migration.php:12]:
```suggestion
```
remove all `// TODO: Implement up() method.`

File: installer/Migration/V4_10_0/Migration.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace OrangeHRM\Installer\Migration\V4_10_0;
+
+class Migration extends \OrangeHRM\Installer\Util\V1\AbstractMigration
```
Comment (RajithaKumara) [inline installer/Migration/V4_10_0/Migration.php:5]: optimize import
```php
use OrangeHRM\Installer\Util\V1\AbstractMigration;

class Migration extends AbstractMigration
```

File: installer/Migration/V4_3_3/Migration.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+class Migration extends \OrangeHRM\Installer\Util\V1\AbstractMigration
+{
+    /**
+     * @inheritDoc
+     */
+    public function up(): void
+    {
+        $this->getSchemaHelper()->changeColumn('hs_hr_employee', 'emp_number', [
+            'Type'=>Types::INTEGER,'Length'=> 7, 'Notnull' => true,'Autoincrement'=>true
```
Comment (RajithaKumara) [inline installer/Migration/V4_3_3/Migration.php:15]:
```suggestion
            'Type' => Types::INTEGER, 'Length' => 7, 'Notnull' => true, 'Autoincrement' => true
```
format the code using IDE

File: installer/Migration/V4_4_0/Migration.php [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+            )
+            ->setParameter('menuTitle', 'Buzz')
+            ->setParameter('screenId', $screenId)
+            ->setParameter('ParentId', null)
+            ->setParameter('level', 1)
+            ->setParameter('orderHint', 1500)
+            ->setParameter('status', 1)
+            ->executeQuery();
+
+        $this->insertConfig('buzz_max_notification_period', '-1 week');
```
Comment (RajithaKumara) [inline installer/Migration/V4_4_0/Migration.php:405]:
```php
// i.e. -4 weeks, -2 days, -1 day, -1 month
// https://www.php.net/manual/en/datetime.formats.relative.php
```
add comment

File: installer/Migration/V4_4_0/Migration.php [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    }
+
+    private function insertConfig(string $key, string $value): void
```
Comment (RajithaKumara) [inline installer/Migration/V4_4_0/Migration.php:408]: doc comment

File: installer/Migration/V4_8_0/Migration.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * @inheritDoc
+     */
+    public function up(): void
+    {
+        $this->insertConfig('help.url', 'https://opensourcehelp.orangehrm.com'); //has access issues.
+        $this->insertConfig('help.processorClass', 'ZendeskHelpProcessor');
+
+        $this->updateConfig('4.8', 'instance.version');
+        $this->updateConfig('80', 'instance.increment_number');
```
Comment (RajithaKumara) [inline installer/Migration/V4_8_0/Migration.php:18]: remove this

File: installer/Migration/V4_8_0/Migration.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+{
+    /**
+     * @inheritDoc
+     */
+    public function up(): void
+    {
+        $this->insertConfig('help.url', 'https://opensourcehelp.orangehrm.com'); //has access issues.
+        $this->insertConfig('help.processorClass', 'ZendeskHelpProcessor');
+
+        $this->updateConfig('4.8', 'instance.version');
```
Comment (RajithaKumara) [inline installer/Migration/V4_8_0/Migration.php:17]: remove this also, will define version in `getVersion`

File: installer/Migration/V4_8_0/Migration.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                    'name' => ':name',
+                    'title' => ':title'
+                ]
+            )
+            ->setParameter('name', 'help')
+            ->setParameter('title', 'Help')
+            ->executeQuery();
+    }
+
+    private function insertConfig(string $value, string $key): void
```
Comment (RajithaKumara) [inline installer/Migration/V4_8_0/Migration.php:33]: doc comment

File: installer/Migration/V4_9_0/Migration.php [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+            ->executeQuery();
+
+        $this->createQueryBuilder()
+            ->delete('ohrm_marketplace_addon')
+            ->andWhere('ohrm_marketplace_addon.plugin_name = :pluginName')
+            ->setParameter('pluginName', 'orangehrmCorporateBrandingPlugin')
+            ->executeQuery();
+
+        $this->updateConfig('4.9', 'instance.version');
+        $this->updateConfig('81', 'instance.increment_number');
```
Comment (RajithaKumara) [inline installer/Migration/V4_9_0/Migration.php:52]: will remove this, remove from other migrations also

File: installer/Migration/V4_9_0/Migration.php [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+            ->executeQuery()
+            ->fetchOne();
+
+        $this->createQueryBuilder()
+            ->insert('ohrm_menu_item')
+            ->values(
+                [
+                    'menu_title' => ':menuTitle',
+                    'screen_id' => ':screenId',
+                    'parent_id' => ':ParentId',
```
Comment (RajithaKumara) [inline installer/Migration/V4_9_0/Migration.php:99]:
```suggestion
                    'parent_id' => ':parentId',
```
minor issue,
`->setParameter('parentId', $parentId)`

File: installer/Migration/V4_9_0/permission/api.yaml [hunk 11]
```diff
@@ -0,0 +1 @@
+
```
Comment (RajithaKumara) [inline installer/Migration/V4_9_0/permission/api.yaml:1]: no need empty files, remove `installer/Migration/V4_9_0/permission/data_group.yaml` also

File: installer/Util/V1/SchemaHelper.php [hunk 12]
```diff
@@ -160,4 +160,12 @@ public function createTable(string $name, string $charset = 'utf8', ?string $col
         $table->setSchemaManager($this->getSchemaManager());
         return $table;
     }
+
+    public function tableExists(array $table): bool
```
Comment (RajithaKumara) [inline installer/Util/V1/SchemaHelper.php:164]: doc comment

File: installer/Util/V1/SchemaHelper.php [hunk 13]
```diff
@@ -160,4 +160,12 @@ public function createTable(string $name, string $charset = 'utf8', ?string $col
...
+    {
+        if ($this->getSchemaManager()->tablesExist($table)) {
+            return true;
+        }
+        return false;
```
Comment (RajithaKumara) [inline installer/Util/V1/SchemaHelper.php:169]:
```suggestion
        return $this->getSchemaManager()->tablesExist($table);
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
good job :tada:

check this https://stackoverflow.com/questions/1580596/how-do-i-make-git-ignore-file-mode-chmod-changes,

File: installer/Migration/V4_10_0/Migration.php [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (SharukaNP) [reply installer/Migration/V4_10_0/Migration.php:2]: added

File: installer/Migration/V4_10_0/Migration.php (hunk 1 above)
Comment (SharukaNP) [reply installer/Migration/V4_10_0/Migration.php:12]: Removed

File: installer/Migration/V4_10_0/Migration.php (hunk 2 above)
Comment (SharukaNP) [reply installer/Migration/V4_10_0/Migration.php:5]: Done

File: installer/Migration/V4_4_0/Migration.php (hunk 4 above)
Comment (SharukaNP) [reply installer/Migration/V4_4_0/Migration.php:405]: Added

File: installer/Migration/V4_4_0/Migration.php (hunk 5 above)
Comment (SharukaNP) [reply installer/Migration/V4_4_0/Migration.php:408]: Added

File: installer/Migration/V4_8_0/Migration.php (hunk 8 above)
Comment (SharukaNP) [reply installer/Migration/V4_8_0/Migration.php:33]: added

File: installer/Migration/V4_9_0/Migration.php (hunk 9 above)
Comment (SharukaNP) [reply installer/Migration/V4_9_0/Migration.php:52]: removed aded to getVersion

File: installer/Migration/V4_8_0/Migration.php (hunk 7 above)
Comment (SharukaNP) [reply installer/Migration/V4_8_0/Migration.php:17]: Removed

File: installer/Migration/V4_9_0/Migration.php (hunk 10 above)
Comment (SharukaNP) [reply installer/Migration/V4_9_0/Migration.php:99]: changed

File: installer/Util/V1/SchemaHelper.php (hunk 12 above)
Comment (SharukaNP) [reply installer/Util/V1/SchemaHelper.php:164]: added

File: installer/Migration/V4_9_0/permission/api.yaml (hunk 11 above)
Comment (SharukaNP) [reply installer/Migration/V4_9_0/permission/api.yaml:1]: removed

File: installer/Util/V1/SchemaHelper.php (hunk 13 above)
Comment (SharukaNP) [reply installer/Util/V1/SchemaHelper.php:169]: changed

File: installer/Migration/V4_8_0/Migration.php (hunk 6 above)
Comment (SharukaNP) [reply installer/Migration/V4_8_0/Migration.php:18]: removed

File: installer/Migration/V4_3_3/Migration.php (hunk 3 above)
Comment (SharukaNP) [reply installer/Migration/V4_3_3/Migration.php]: FORMATTED

File: installer/Migration/V5_0_0_beta/Migration.php [hunk 15]
```diff
@@ -392,6 +392,14 @@ public function up(): void
             ->executeQuery();
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function getVersion(): string
+    {
+        return '5.0.0.beta';
```
Comment (RajithaKumara) [inline installer/Migration/V5_0_0_beta/Migration.php:400]:
```suggestion
        return '5.0-beta';
```

File: installer/Migration/V4_10_0/Migration.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        // no db changes in this version
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getVersion(): string
+    {
+        return '4.10.0';
```
Comment (RajithaKumara) [inline installer/Migration/V4_10_0/Migration.php:39]:
```suggestion
        return '4.10';
```
no need to change namespace

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**